File: count-number-of-homogenous-substrings/count-number-of-homogenous-substrings.go
func countHomogenous(s string) int {
    currSize,ans := 0,0
    mod := 1000000007
    length := len(s)
    currChar := byte('a')
    for i:= 0 ; i < length ; i++ {
        if(s[i] == currChar){
            currSize+=1
        }else{
            ans = (ans%mod + (currSize*(currSize+1)/2)%mod)%mod
            currSize = 1
            currChar = s[i]
        }
    }
    ans = (ans%mod + (currSize*(currSize+1)/2)%mod)%mod
    return ans
    
}

File: count-number-of-homogenous-substrings/count-number-of-homogenous-substrings.cpp
class Solution {
public:
    int countHomogenous(string s) {
        long ans = 0;
        char curr_char = 'a';
        int i = 0;
        long curr_size = 0;
        long mod = 1000000007;
        while(i < s.size()){
            if(s[i] == curr_char){
                curr_size++;
            }else{
                ans= (ans%mod + (curr_size*(curr_size+1)/2)%mod)%mod ;
                curr_size = 1;
                curr_char = s[i];
            }
            i++;
        }
        ans= (ans%mod + (curr_size*(curr_size+1)/2)%mod)%mod ;
        return (int)ans;
    }
};